test/test_compile_package.py

```diff
@@ -0,0 +1,374 @@
+        self._test_load(f, args, expected, self.path())
+
+        f(*args1)
+        self._test_save(f, args1, expected1, self.path(), 1)
+        self._test_load(f, args1, expected1, self.path())
+
+    @requires_cuda
+    @parametrize("device", ["cpu", "cuda"])
+    def test_basic_module_forward_static(self, device):
+        torch.set_default_device(device)
+
+        class Module(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.register_buffer("p", torch.tensor(10))
+
+            def forward(self, a, b, c):
+                assert c is None
+                for i in range(30):
+                    a = a + b * i + self.p
+                return a
+
+        f = Module()
+        args = torch.randn(3), torch.randn(3), None
+        args1 = torch.randn((4, 2)), torch.randn((4, 2)), None
+        expected = f(*args)
+        expected1 = f(*args1)
+        f = torch.compile(f, fullgraph=True, dynamic=False, package=self.path())
+        f(*args)
+        self._test_save(f, args, expected, self.path(), 0)
+        self._test_load(f, args, expected, self.path())
+
+        f(*args1)
+        self._test_save(f, args1, expected1, self.path(), 1)
+        self._test_load(f, args1, expected1, self.path())
+
+    @requires_cuda
+    @parametrize("device", ["cpu", "cuda"])
+    def test_basic_method_forward_static(self, device):
+        torch.set_default_device(device)
+
+        m = ClassForTest()
+        args = torch.randn(3), torch.randn(3)
+        args1 = torch.randn((4, 2)), torch.randn((4, 2))
+        expected = m.g(*args)
+        expected1 = m.g(*args1)
+        f = torch.compile(m.g, fullgraph=True, dynamic=False, package=self.path())
+        f(*args)
+        self._test_save(f, args, expected, self.path(), 0)
+        self._test_load(f, args, expected, self.path())
+
+        f(*args1)
+        self._test_save(f, args1, expected1, self.path(), 1)
+        self._test_load(f, args1, expected1, self.path())
+
+    def test_basic_class_method(self):
+        class Module:
+            def __init__(self):
+                self.p = torch.tensor(10)
+
+            def g(self, a, b):
+                for i in range(30):
+                    a = a + b * i + self.p
+                return a
+
+        m = Module()
+        args = torch.randn(3), torch.randn(3)
+        f = torch.compile(m.g, fullgraph=True, dynamic=False, package=self.path())
+        with self.assertRaisesRegex(
+            RuntimeError, "Please define the class at global scope"
+        ):
+            f(*args)
+
+    @requires_cuda
+    @parametrize("device", ["cpu", "cuda"])
+    def test_basic_function_forward_non_fullgraph(self, device):
+        torch.set_default_device(device)
+
+        def f(a, b):
+            for i in range(30):
+                a = a + b * i
+            return a
+
+        with self.assertRaisesRegex(
+            RuntimeError, "Compile package is only supported .*fullgraph=True"
```

Comment:
I want to confirm you plan to remove the limitation, since a lot of key users of this have graph breaks.